### ISSUE TITLE
Callable detection with a Union containing a type variable

### ISSUE THREAD

**Issue by jparise** [microsoft/pyright#1748]
**Describe the bug**
A `Union`-typed value that contains a `Callable` is not determined to be callable after a `callable()` check.

```py
T = TypeVar('T')

def test(arg: Union[T, Callable[[], T]]) -> T:
    if callable(arg):
        return arg()
    return arg
```

```
  foo.py:12:16 - error: Object of type "T@test" is not callable (reportGeneralTypeIssues)
  foo.py:12:16 - error: Expression of type "Unknown | T@test" cannot be assigned to return type "T@test"
    Type "Unknown | T@test" cannot be assigned to type "T@test" (reportGeneralTypeIssues)
```

(Mypy doesn't produce an error in this case.)

When I drop the type variable in favor if a simple type, there is no error:

```py
def test(arg: Union[int, Callable[[], int]]) -> int:
    if callable(arg):
        return arg()
    return arg
```

**VS Code extension or command-line**
pyright 1.1.129

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in Pyright 1.1.130, which I just published. It will also be part of the next release of Pylance.
